crates/bevy_pbr/src/render/utils.wgsl

```diff
@@ -68,6 +68,16 @@ fn octahedral_decode_signed(v: vec2<f32>) -> vec3<f32> {
     return normalize(n);
 }
 
+// https://jcgt.org/published/0006/01/01/paper.pdf
+fn build_orthonormal_basis(normal: vec3<f32>) -> mat3x3<f32> {
+    let sign = select(-1.0, 1.0, normal.z >= 0.0);
+    let a = -1.0 / (sign + normal.z);
+    let b = normal.x * normal.y * a;
+    let tangent = vec3(1.0 + sign * normal.x * normal.x * a, sign * b, -sign * normal.x);
+    let bitangent = vec3(b, sign + normal.y * normal.y * a, -normal.y);
+    return mat3x3(tangent, bitangent, normal);
+}
+
```

Comment:
I'd wait for https://github.com/bevyengine/bevy/pull/20191 and then rebase - we have a bunch of duplicates of this code already

Reply:
Instead of blocking on this PR on #20191 , I could just take a copy of this function into the shader for now and revert the changes for reusing it. and then in a second PR - since we are already trying to cut scope, move things to follow up - I will address the regenerate flag and reusing functions as much as possible. what do you think?